doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1836 @@
+might allow other manipulations to work. It is usually a good idea to be as
+precise as possible about the assumptions on a symbol when creating it.
+
+
+The (old) assumptions system
+============================
+
+There are two sides to the assumptions system. The first side is that we can
+declare assumptions on a symbol when creating the symbol. The other side is
+that we can query the assumptions on any expression using the corresponding
+``is_*`` attribute. For example:
+
+    >>> x = Symbol('x', positive=True)
+    >>> x.is_positive
+    True
+
+We can query assumptions on any expression not just a symbol:
+
+    >>> expr = 1 + x**2
+    >>> expr
+    x**2 + 1
+    >>> expr.is_positive
+    True
+    >>> expr.is_negative
+    False
+
+The values given in an assumptions query use three-valued "fuzzy" logic. Any
+query can return ``True``, ``False``, or ``None`` where ``None`` should be
+interpreted as meaning that the result is *unknown*.
+
+    >>> x = Symbol('x')
+    >>> y = Symbol('y', positive=True)
+    >>> z = Symbol('z', negative=True)
+    >>> print(x.is_positive)
+    None
+    >>> print(y.is_positive)
+    True
+    >>> print(z.is_positive)
+    False
+
+.. note:: We need to use ``print`` in the above examples because the special
+          value ``None`` does not display by default in the Python
+          interpretter.
+
+There are two reasons why an assumptions query might give ``None``. It is
+possible that the query is *unknowable* as in the case of ``x`` above. Since
+``x`` does not have any assumptions declared it roughly represents an
+arbitrary complex number. An arbitrary complex number *might* be a positive
+real number but it also might *not* be. Without further information there is
+no way to resolve the query ``x.is_positive``.
+
+The other reason that an assumptions query might give ``None`` is just that
+the assumptions system does not try very hard to answer complicated queries.
+The system is intended to be fast and uses simple heuristic methods to
+conclude a ``True`` or ``False`` answer in common cases. For example any sum
+of positive terms is positive so:
+
+    >>> from sympy import symbols
+    >>> x, y = symbols('x, y', positive=True)
+    >>> expr = x + y
+    >>> expr
+    x + y
+    >>> expr.is_positive
+    True
+
+The last example is particularly simple so the assumptions system is able to
+give a definite answer. If the sum involved a mix of positive or negative
+terms it would be a harder query:
+
+    >>> x = Symbol('x', real=True)
+    >>> expr = 1 + (x - 2)**2
+    >>> expr
+    (x - 2)**2 + 1
+    >>> expr.is_positive
+    True
+    >>> expr2 = expr.expand()
+    >>> expr2
+    x**2 - 4*x + 5
+    >>> print(expr2.is_positive)
+    None
+
+Ideally that last example would give ``True`` rather than ``None`` because the
+expression is always positive for any real value of ``x`` (and ``x`` has been
+assumed real). The assumptions system is intended to be efficient though: it
```

Comment:
It's really three things: some algorithms aren't implemented because they would be too slow, some aren't implemented because they just aren't implemented yet, and some cases simply don't have algorithms at all (many assumptions are in general computationally undecidable).

Reply:
I'd make it four things:

1. Some queries are unknowable because there is not enough information provided e.g. `Symbol('x', real=True).is_positive`.

2. For some queries sufficient information is provided that a definite answer exists but the algorithmic problem of resolving the question for the class of queries is undecidable (Richardson's theorem).

3. For some queries algorithms exist but they are too slow and are deliberately not implemented.

4. In some cases simple algorithms could resolve the query but they haven't yet been implemented in sympy.